### ISSUE TITLE
[Enhancement] Hide the tree view if non-supported file type is active

### ISSUE THREAD

**Issue by kubawolanin** [DaGhostman/vscode-tree-view#36]
Hello @DaGhostman,

Currently if I'm working on e.g. a SCSS file, the tree view is still there, but empty:
![image](https://user-images.githubusercontent.com/2270505/35484736-7b84685a-0454-11e8-8c07-09aa41c7bb18.png)

Would be nice if the extension activated the view only if we're editing a supported file (*.rules, *.java, *.php etc).

FWIW you can specify a `resourceLangId` property in `package.json` in `when` condition to achieve that, e.g.:

```json
        "views": {
            "explorer": [
                {
                    "id": "openhabItems",
                    "name": "openHAB Items",
                    "when": "resourceLangId == openhab"
                }
            ]
        },
```

So in this case I think it would be:
```json
    "contributes": {
        "views": {
            "explorer": [
                {
                    "id": "tree-outline",
                    "name": "Tree",
                    "when": "resourceLangId == php || resourceLangId == json || resourceLangId == js || resourceLangId == openhab" // etc...
                }
            ]
        },
```

As you can see it will suddenly become a quite big list so I wonder if we could somehow split that up.

Cheers!

**Comment by DaGhostman** [comment]
actually, I tried that approach initially, but it didn't work (it didn't now as well) it seems that it is expecting to see a single condition. I would love to see that working or some other flexible approach but as it is there is none so far. This will be achievable if multiple `views` are each one with it's own support and that should work ideally.

However, I've noticed another weird behavior that I will investigate further and possible open a vscode issue for more information.
